python/vowpalwabbit/pyvw.py

```diff
@@ -1308,25 +1305,24 @@ class CCBLabel(AbstractLabel):
 
     def __init__(
         self,
-        type: Union["example", CCBLabelType] = CCBLabelType.UNSET,
+        type: CCBLabelType = CCBLabelType.UNSET,
         explicit_included_actions: List[int] = [],
         weight: float = 1,
         outcome: Optional[CCBSlotOutcome] = None,
     ):
         AbstractLabel.__init__(self)
-        if isinstance(type, example):
-            self.from_example(type)
-        else:
-            self.type = type
-            self.explicit_included_actions = explicit_included_actions
-            self.weight = weight
-            self.outcome = outcome
-
-    def from_example(self, ex: "example"):
-        self.type = ex.get_ccb_type()
-        self.explicit_included_actions = ex.get_ccb_explicitly_included_actions()
-        self.weight = ex.get_ccb_weight()
-        self.outcome = CCBSlotOutcome(ex) if self.type == CCBLabelType.SLOT else None
+        self.type = type
+        self.explicit_included_actions = explicit_included_actions
+        self.weight = weight
+        self.outcome = outcome
+
+    @staticmethod
+    def from_example(ex: "Example"):
+        type = ex.get_ccb_type()
+        explicit_included_actions = ex.get_ccb_explicitly_included_actions()
+        weight = ex.get_ccb_weight()
+        outcome = CCBSlotOutcome.from_example(ex) if type == CCBLabelType.SLOT else None
```

Comment:
You only want to set an outcome if one exists, not if it is a slot example. The c++ should project a `has_outcome` sort of function so that can be checked

Reply:
Question: Only slot type has outcome. For the slots which don't it can be an empty list inside `CCBSlotOutcome`.

Reply:
Yes but a slot is not guaranteed to have an outcome. There being an outcome or  not is actually an important distinction for two given labels, even if they are both slots

Reply:
Okay, also added a `slot pred`  to test which confirms the following behaviour

`ccb slot |` type of example should have no outcome even though its a slot type.